### ISSUE TITLE
Building wheel fails with latest release on all platforms which lack wheels

### ISSUE THREAD

**Issue by dan-blanchard** [ParallelSSH/parallel-ssh#98]
__Steps to reproduce:__

1. `pip install parallel-ssh` from a platform for which wheels are not on PyPI
2. Building the wheel fails with: 

```
  Failed building wheel for parallel-ssh
  Running setup.py clean for parallel-ssh
Failed to build parallel-ssh
Installing collected packages: parallel-ssh, setuptools, certifi, urllib3, chardet, ply
  Found existing installation: parallel-ssh 1.2.0
    Uninstalling parallel-ssh-1.2.0:
      Successfully uninstalled parallel-ssh-1.2.0
  Running setup.py install for parallel-ssh ... error
    Complete output from command /Users/dan/.pyenv/versions/miniconda3-latest/envs/streamparse3/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/h6/389yk8291lj2h9q9020105p00000gn/T/pip-build-h0d9xlxw/parallel-ssh/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /var/folders/h6/389yk8291lj2h9q9020105p00000gn/T/pip-c4cxfxds-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.9-x86_64-3.6
    creating build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/pssh2_client.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/_version.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/constants.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/__init__.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/ssh2_client.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/utils.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/agent.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/pssh_client.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/exceptions.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/base_pssh.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/ssh_client.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    copying pssh/output.py -> build/lib.macosx-10.9-x86_64-3.6/pssh
    creating build/lib.macosx-10.9-x86_64-3.6/pssh/native
    copying pssh/native/__init__.py -> build/lib.macosx-10.9-x86_64-3.6/pssh/native
    running build_ext
    building 'pssh.native._ssh2' extension
    creating build/temp.macosx-10.9-x86_64-3.6
    creating build/temp.macosx-10.9-x86_64-3.6/pssh
    creating build/temp.macosx-10.9-x86_64-3.6/pssh/native
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/dan/.pyenv/versions/miniconda3-latest/envs/streamparse3/include -arch x86_64 -I/Users/dan/.pyenv/versions/miniconda3-latest/envs/streamparse3/include -arch x86_64 -Ilibssh2/include -I/Users/dan/.pyenv/versions/miniconda3-latest/envs/streamparse3/include/python3.6m -c pssh/native/_ssh2.c -o build/temp.macosx-10.9-x86_64-3.6/pssh/native/_ssh2.o -O3
    clang: error: no such file or directory: 'pssh/native/_ssh2.c'
    clang: error: no input files
    error: command 'gcc' failed with exit status 1
```
__Expected behaviour:__ Install succeeds

__Actual behaviour:__  Crash illustrated above

__Additional info:__ ssh2-python==0.7.0.post4, clang/llvm 9.0.0

**Comment by dan-blanchard** [comment]
I've seen it on macOS 10.13.2 with CPython and on [Travis with PyPy](https://travis-ci.org/Parsely/streamparse/jobs/322946368).

**Comment by dan-blanchard** [comment]
I should note that the earlier stack trace is if you don't have Cython installed, probably because the package doesn't include the C file. If you have Cython installed you get a much longer series of compilation errors as shown in [this gist](https://gist.github.com/dan-blanchard/c03113fc949e1a30b36bc063060c3f94).

**Comment by pkittenis** [comment]
Hi there,

Thanks for the report! This is resolved as of `1.2.1-post2`, per below:

```
pip install -U --no-use-wheel parallel-ssh
<..>
Skipping bdist_wheel for parallel-ssh, due to binaries being disabled for it.
Installing collected packages: parallel-ssh
  Running setup.py install for parallel-ssh ... done
Successfully installed parallel-ssh-1.2.1.post2
```

Note that `libssh2` headers are required as well for building `parallel-ssh`.  For conda environments, [there is a conda package provided](http://ssh2-python.readthedocs.io/en/latest/installation.html#conda-package) for `ssh2-python` that will install both it and `libssh2`.

There was a separate issue with building OSX wheels - 10.12 should have had a wheel built for it already - that has also been resolved.
